### PR TITLE
Update MSRV to 1.65

### PR DESCRIPTION
**Pull request by russell-islam**
Sev-Snp on MSHV uses igvm crate to parse the igvm file. igvm crate needs minimum rust version 1.65 to build.

### CODE REVIEW

**Comment by jinankjain** [comment]
There is a typo in the commit message vertsion -> version. Apart from that looks good to me.

**Comment by russell-islam** [comment]
> There is a typo in the commit message vertsion -> version. Apart from that looks good to me.

fixed

**Review by jinankjain — APPROVED** [review]

**Comment by rbradford** [comment]
The commit message should say "Update MSRV to 1.65" - we have other versions elsewhere (like the Dockerfile.)

**Comment by russell-islam** [comment]
> The commit message should say "Update MSRV to 1.65" - we have other versions elsewhere (like the Dockerfile.)

Docker file already has version 1.67.1 as per this line: RUST_TOOLCHAIN="1.67.1". Do we still need to change it?

**Comment by likebreath** [comment]
> > The commit message should say "Update MSRV to 1.65" - we have other versions elsewhere (like the Dockerfile.)
> 
> Docker file already has version 1.67.1 as per this line: RUST_TOOLCHAIN="1.67.1". Do we still need to change it?

That's exactly Rob's point. Your commit message subject is ambiguous. 

Try: `build: Bump MSRV to 1.65`.

File: virtio-devices/src/block.rs [hunk 1]
```diff
@@ -616,7 +616,9 @@ impl Block {
                 (disk_nsectors, avail_features, 0, config, false)
             };
 
-        let serial = serial.map(Vec::from).unwrap_or(build_serial(&disk_path));
+        let serial = serial
+            .map(Vec::from)
+            .unwrap_or_else(|| build_serial(&disk_path));
```
Comment (likebreath) [inline virtio-devices/src/block.rs:621]: Also, why these changes are needed? If they are for accommodating clippy warnings, please put it in a separate commit.

Comment (russell-islam) [reply virtio-devices/src/block.rs:621]: Done.

Comment (rbradford) [reply virtio-devices/src/block.rs:621]: It's very strange that this wasn't found by the stable/beta/nightly runs... :confused:

**Review by rbradford — APPROVED** [review]
